Core/include/Acts/Vertexing/AdaptiveMultiVertexFitter.ipp

```diff
@@ -37,66 +37,71 @@ Acts::AdaptiveMultiVertexFitter<input_track_t, linearizer_t>::fitImpl(
   // Reset annealing tool
   state.annealingState = AnnealingUtility::State();
 
-  // Indicates how much the vertex positions have shifted
-  // in last fit iteration. Will be false if vertex position
-  // shift was too big. Needed if equilibrium is reached in
-  // annealing procedure but fitter has not fully converged
-  // yet and needs some more iterations until vertex position
-  // shifts between iterations are small (converged).
+  // Boolean indicating whether any of the vertices has moved more than
+  // m_cfg.maxRelativeShift during the last iteration. We will keep iterating
+  // until the equilibrium (i.e., the lowest temperature) is reached in
+  // the annealing procedure and isSmallShift is true (or until the maximum
+  // number of iterations is exceeded).
   bool isSmallShift = true;
 
   // Number of iterations counter
   unsigned int nIter = 0;
 
   // Start iterating
-  while (nIter < m_cfg.maxIterations &&
-         (!state.annealingState.equilibriumReached || !isSmallShift)) {
+  while (nIter < m_cfg.maxIterations and
+         (not state.annealingState.equilibriumReached or not isSmallShift)) {
     // Initial loop over all vertices in state.vertexCollection
-    for (auto currentVtx : state.vertexCollection) {
-      VertexInfo<input_track_t>& currentVtxInfo = state.vtxInfoMap[currentVtx];
-      currentVtxInfo.relinearize = false;
+    for (auto vtx : state.vertexCollection) {
+      VertexInfo<input_track_t>& vtxInfo = state.vtxInfoMap[vtx];
+      vtxInfo.relinearize = false;
       // Store old position of vertex, i.e. seed position
       // in case of first iteration or position determined
       // in previous iteration afterwards
-      currentVtxInfo.oldPosition = currentVtx->fullPosition();
-
-      Vector4 dist = currentVtxInfo.oldPosition - currentVtxInfo.linPoint;
-      double perpDist = std::hypot(dist[0], dist[1]);
-      // Determine if relinearization is needed
-      if (perpDist > m_cfg.maxDistToLinPoint) {
-        // Relinearization needed, distance too big
-        currentVtxInfo.relinearize = true;
-        // Prepare for fit with new vertex position
-        prepareVertexForFit(state, currentVtx, vertexingOptions);
+      vtxInfo.oldPosition = vtx->fullPosition();
+
+      // Calculate the x-y-distance between the current vertex position
+      // and the linearization point of the tracks. If it is too large,
+      // we relinearize the tracks and recalculate their 3D impact
+      // parameters.
+      ActsVector<2> xyDiff = vtxInfo.oldPosition.template head<2>() -
+                             vtxInfo.linPoint.template head<2>();
+      double xyDist = std::hypot(xyDiff[0], xyDiff[1]);
```

Comment:
Technically, there's also `xyDiff.norm()` or `xyDiff.normSquared()` if you want to skip the sqrt.